sdks/python/apache_beam/io/gcp/gcsio_test.py

```diff
@@ -542,10 +542,10 @@ def test_file_iterator(self):
     line_count = 10
     for _ in range(line_count):
       line_length = random.randint(100, 500)
-      line = os.urandom(line_length).replace('\n', ' ') + '\n'
+      line = os.urandom(line_length).replace(b'\n', b' ') + b'\n'
       lines.append(line)
 
-    contents = ''.join(lines)
+    contents = b''.join(lines)
```

Comment:
Why converting to bytes works here? Can you explain a little more about the problem?

Reply:
`os.urandom` returns a `bytestring`. On Python 2, this worked using `string`, since a Python 2 `string` is actually a `bytestring` (`'' = b''`), while on Python 3 a `string` is actually a `unicode` string (`'' = u''`). By defining `b''`, we get the same behavior on both Python versions.